### ISSUE TITLE
Unused Assignment false positive

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#33949]
https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs#L3893

This is flagged as an unused assignment, but it is used.

Update: I got a minimal repro, which requires a conditional expression as an argument prior to the read.

```csharp
class A
{
    public static void M(int? x)
    {
        A a = new A();
        a = M2(x ?? 1, a);
    }

    private static A M2(int? x, A a)
    {
        return a;
    }
}
```
